src/generator.ts

```diff
@@ -1006,15 +1006,46 @@ export class Generator {
     if (!install) {
       await this.traceMap.processInputMap;
       return this.install(
-        Object.keys(this.traceMap.installer.installs.primary)
+        Object.entries(this.traceMap.installer.installs.primary).map(
+          ([alias, target]) => {
+            const pkgTarget =
+              this.traceMap.installer.constraints.primary[alias];
+
+            // Try to reinstall lock against constraints if possible, otherwise
```

Comment:
Idea is that if there's a constraint for a given alias mapping, we reinstall against the package for that constraint, rather than against the alias name (since that would just clobber the alias). If there's no such constraint, we just treat it as a custom mapping and reinstall against the URL.